src/app/core/services/rundown-state.service.ts

```diff
@@ -75,16 +74,34 @@ export class RundownStateService implements OnDestroy {
   }
 
   private activateRundownFromEvent(event: RundownActivatedEvent): void {
-    const rundownSubject = this.rundownSubjects.get(event.rundownId)
+    const rundownSubject = this.getRundownSubject(event.rundownId)
     if (!rundownSubject) {
       return
     }
     const activeRundown: Rundown = this.rundownEntityService.activate(rundownSubject.value)
     rundownSubject.next(activeRundown)
   }
 
+  private getRundownSubject(rundownId: string): BehaviorSubject<Rundown> | undefined {
+    const rundownSubject = this.rundownSubjects.get(rundownId)
+    if (!rundownSubject) {
+      return
+    }
+    const wasRemoved: boolean = this.removeSubjectIfHasNoObservers(rundownSubject).wasRemoved
+    return wasRemoved ? undefined : rundownSubject
+  }
+
+  private removeSubjectIfHasNoObservers(rundownSubject: BehaviorSubject<Rundown>): { wasRemoved: boolean } {
```

Comment:
The name of this method sounds weird in my mind. Propably because the sentence it almost writes are missing a single word. Sounds like it is missing an 'It' between 'If' and 'Has'. I understand why you didn't add that, as the name is already quite long.

Would the name `removeObserverlessSubject` maybe be better?

Reply:
`RundownTimingContextStateService.removeSubjectIfHasNoObserversOrSubscriptions` is similarly named, thus might benefit form similar renaming.

Reply:
`removeObserverlessSubject` sounds like it expects a subject that has no observers and will always remove it. The `If` is there to indicate the conditional effect. I have updated it with an `It` inserted.